File: backend/src/main/scala/ch/countrygame/Classes/Relation.scala
package ch.countrygame.Classes

import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

case class Relation(id: Int, countryId1: Int, countryId2: Int, relationType: Int) {
}
object Relation {
  implicit val encoder = deriveEncoder[Relation]
  implicit val decoder = deriveDecoder[Relation]
}

File: frontend/src/app/models/relation.ts
export class Relation {
  id: number;
  countryId1: number;
  countryId2: number;
  relationType: number;

  constructor(id: number, countryId1: number, countryId2: number, relationType: number) {
    this.id = id;
    this.countryId1 = countryId1;
    this.countryId2 = countryId2;
    this.relationType = relationType;
  }
}
